.azure-pipelines/package_nightly_and_release.yml

```diff
@@ -47,6 +47,9 @@ extends:
       windows:
         py_versions: ['3.7', '3.8', '3.9']
         conda_env: 'scipp-developer-minimal.yml'
+      windows-debug:
+        py_versions: ['3.7', '3.8', '3.9']
+        conda_env: 'scipp-developer-minimal.yml'
```

Comment:
Which conda label will this deploy to?

Reply:
As far as I can tell so far, all architectures are deploying into a conda label called `test`, unless I'm just missing where it's overridden. Does it make sense to create a `test-debug` label or similar to hold our debug builds?

Secondary question: do we also want nightly debug builds on linux/osx as well as windows?

Reply:
It should deploy to `dev` (see a couple lines below). But maybe we should split the build config discussion/changes from the fixes of the debug build? Can you remove all CI related changes from this PR, so we can merge the code fixes. We can then handle potential CI changes independently.

Reply:
Ok, that's fine. I'll remove the CI changes now.